llvm/include/llvm/SYCLLowerIR/IR/PassAdaptors.h

```diff
@@ -0,0 +1,72 @@
+//===--------------- PassAdaptors.h - SYCL Pass Adaptors ----------------- ===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+/// \file
+/// This header defines Pass Adaptors which allows to run Function Passes
+/// on a subset of functions from given Module.
+/// LLVM does not provide a default utility for this purpose.
+//===----------------------------------------------------------------------===//
+
+#include "llvm/IR/PassManager.h"
+#include "llvm/IR/PassManagerInternal.h"
+#include "llvm/Support/raw_ostream.h"
+#include <memory>
+
+#ifndef LLVM_SYCL_IR_PASS_ADAPTORS
+#define LLVM_SYCL_IR_PASS_ADAPTORS
+
+namespace llvm {
+
+/// Trivial adaptor that maps from a module to functions marked
+/// by "sycl-framework" metadata. The implementation is mostly a copy-paste
+/// of LLVM ModuleToFunctionPassAdaptor class.
+///
+/// Designed to allow composition of a FunctionPass(Manager) and
+/// a ModulePassManager, by running the FunctionPass(Manager) over every
+/// "sycl-framework "function in the module.
+class ModuleToSYCLFrameworkFunctionPassAdaptor
```

Comment:
I added here 2 files in `SYCLLowerIR/IR/PassAdaptors.{h,cpp}` which copies a functionality from `llvm/IR/PassManager.{h,cpp}` of a class `ModuleToFunctionPassAdaptor` but adds a modification that we run a pass on a Function which is marked with `sycl-framework` metadata.

Are we OK with such approach?